src/main/java/com/google/sps/data/MemeDatabase.java

```diff
@@ -0,0 +1,62 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import java.util.ArrayList;
+import java.util.List;
+
+// Database for all memes to be accessed
+public class MemeDatabase {
+  private DatastoreService datastore;
+  private Query query;
+  public static final String ENTITY_QUERY_STRING = "meme";
+  public static final String TITLE_QUERY_STRING = "title";
+  public static final String SOURCE_QUERY_STRING = "source";
+  public static final String ID_QUERY_STRING = "id";
+
+  public MemeDatabase(DatastoreService datastore) {
+    this.datastore = datastore;
+    query = new Query(ENTITY_QUERY_STRING);
+  }
+
+  public void addMemeToDatabase(Meme meme) {
+    try {
+      Entity memeEntity = memeToEntity(meme);
+      datastore.put(memeEntity);
+    } catch (IllegalArgumentException e) {
+      e.printStackTrace();
```

Comment:
Probably need to report something to the user? We're mostly ignoring this exception right now.